File: LeetCode/1720解码异或后的数组.cpp
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* decode(int* encoded, int encodedSize, int first, int* returnSize){
    int *res = (int *)malloc(sizeof(int) * (encodedSize + 5));
    *returnSize = 0;
    res[(*returnSize)++] = first;
    for (int i = 0; i < encodedSize; i++) {
        res[(*returnSize)++] = encoded[i] ^ first;
        first = res[i + 1];
    }
    return res;
}

File: LeetCode/1720解码异或后的数组.go
func decode(encoded []int, first int) []int {
    var res = make([]int, len(encoded) + 1)
    res[0] = first
    for i := 0; i < len(encoded); i++ {
        res[i + 1] = encoded[i] ^ res[i]
    }
    return res
}